### DISCUSSION TITLE
Is there a way to pass parameters to the query as an array/list/tuple?

### DISCUSSION THREAD

**Discussion by ameetnsharma** [sfu-db/connector-x#299] [General]
So this is the type of thing I'm doing:

df = cx.read_sql(f"mysql://{db_username}:{db_password}@{db_host}:3306/{db_name}", query)
        
where the query is just a raw text query. Works well, but my work superiors are concerned with sql injection and want to be able to pass parameters to the query as a tuple or list... like here:

https://www.psycopg.org/psycopg3/docs/basic/params.html

Is this possible with connectorx? Thanks.

**Comment by wangxiaoying** [comment]
Hi @ameetnsharma , unfortunately we are now not supporting this feature. The implementation might not be trivial since we need to figure out the type of the parameters from python and pass it to different rust drivers. Anyway, thanks for the suggestion and welcome to help! : )

**Reply by saolof** [reply]
@wangxiaoying  What about supporting the v2.0 query object from sqlalchemy core as an input type? This is a dealbreaker for us too when considering this library. Malicious attacks are one thing, but so are french place names with quotes in them
